java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java

```diff
@@ -57,25 +60,48 @@
 
     private static final String TEST_DATA = "EPISODEII_IS_BEST_EPISODE";
 
+    private Boolean bucketsEnabled = false;
+
     private MantaClient mantaClient;
 
     private String testPathPrefix;
 
+    private String bucketsTestPathPrefix;
+
     @BeforeClass
-    @Parameters({"usingEncryption"})
-    public void beforeClass(@Optional Boolean usingEncryption) throws IOException {
+    @Parameters({"usingEncryption", "testType"})
+    public void beforeClass(@Optional Boolean usingEncryption,
+                            @Optional String testType) throws IOException {
 
         // Let TestNG configuration take precedence over environment variables
         ConfigContext config = new IntegrationTestConfigContext(usingEncryption);
+        String testName = this.getClass().getSimpleName();
 
         mantaClient = new MantaClient(config);
-        testPathPrefix = IntegrationTestConfigContext.generateBasePath(config, this.getClass().getSimpleName());
-        mantaClient.putDirectory(testPathPrefix, true);
+
+        if(testType != null && testType.equals("buckets")) {
+            if(IntegrationTestHelper.verifyBucketsSupport(config, mantaClient)) {
+                bucketsEnabled = true;
```

Comment:
Could all of this logic be put in a helper? Does it have to be duplicated in every integration test?